test/performance/odp_bench_event_vector.c

```diff
@@ -0,0 +1,760 @@
+	odp_event_t *event_tbl = gbl_args->event_tbl;
+	int i;
+
+	for (i = 0; i < TEST_REPEAT_COUNT; i++)
+		evv_arr[i] = odp_event_vector_from_event(event_tbl[i]);
+
+	return i;
+}
+
+static int event_vector_to_event(void)
+{
+	odp_event_vector_t *evv_arr = gbl_args->evv_arr;
+	odp_event_t *event_tbl = gbl_args->event_tbl;
+	int i;
+
+	for (i = 0; i < TEST_REPEAT_COUNT; i++)
+		event_tbl[i] = odp_event_vector_to_event(evv_arr[i]);
+
+	return i;
+}
+
+static int event_vector_size(void)
+{
+	odp_event_vector_t *evv_arr = gbl_args->evv_arr;
+	uint32_t ret = 0;
+
+	for (int i = 0; i < TEST_REPEAT_COUNT; i++)
+		ret += odp_event_vector_size(evv_arr[i]);
+
+	return ret;
+}
+
+static int event_vector_user_area(void)
+{
+	odp_event_vector_t *evv_arr = gbl_args->evv_arr;
+	void **ptr_tbl = gbl_args->ptr_tbl;
+	int i;
+
+	for (i = 0; i < TEST_REPEAT_COUNT; i++)
+		ptr_tbl[i] = odp_event_vector_user_area(evv_arr[i]);
+
+	return i;
+}
+
+static int event_vector_pool(void)
+{
+	odp_event_vector_t *evv_arr = gbl_args->evv_arr;
+	odp_pool_t *pool_tbl = gbl_args->pool_tbl;
+	int i;
+
+	for (i = 0; i < TEST_REPEAT_COUNT; i++)
+		pool_tbl[i] = odp_event_vector_pool(evv_arr[i]);
+
+	return i;
+}
+
+static int event_vector_alloc(void)
+{
+	odp_event_vector_t *evv_arr = gbl_args->evv_arr;
+	odp_pool_t evv_pool = gbl_args->evv_pool;
+	odp_pool_t buf_pool = gbl_args->buf_pool;
+	uint32_t evv_size = gbl_args->evv_size;
+	odp_buffer_t buf[gbl_args->evv_size];
+	odp_event_t *tbl;
+	int i;
+
+	for (i = 0; i < TEST_REPEAT_COUNT; i++) {
+		evv_arr[i] = odp_event_vector_alloc(evv_pool);
+		odp_event_vector_tbl(evv_arr[i], &tbl);
+		odp_buffer_alloc_multi(buf_pool, buf, evv_size);
+		odp_buffer_to_event_multi(buf, tbl, evv_size);
+	}
+
+	return i;
+}
+
+static int event_vector_free(void)
+{
+	odp_event_vector_t *evv_arr = gbl_args->evv_arr;
+	uint32_t evv_size = gbl_args->evv_size;
+	odp_event_t *tbl;
+	int i;
+
+	for (i = 0; i < TEST_REPEAT_COUNT; i++) {
+		odp_event_vector_tbl(evv_arr[i], &tbl);
+		odp_event_free_multi(tbl, evv_size);
+		odp_event_vector_free(evv_arr[i]);
+	}
+
+	return i;
+}
+
+static int event_free(void)
+{
+	odp_event_t *event_tbl = gbl_args->event_tbl;
+	int i;
+
+	for (i = 0; i < TEST_REPEAT_COUNT; i++)
+		odp_event_free(event_tbl[i]);
+
+	return i;
+}
+
+static int event_vector_alloc_free(void)
+{
+	odp_pool_t evv_pool = gbl_args->evv_pool;
+	odp_pool_t buf_pool = gbl_args->buf_pool;
+	uint32_t evv_size = gbl_args->evv_size;
+	odp_buffer_t buf[evv_size];
+	odp_event_t *tbl;
+	int i;
+
+	for (i = 0; i < TEST_REPEAT_COUNT; i++) {
+		odp_event_vector_t evv = odp_event_vector_alloc(evv_pool);
+
+		odp_event_vector_tbl(evv, &tbl);
+		int num = odp_buffer_alloc_multi(buf_pool, buf, evv_size);
+
+		if (odp_unlikely(num < 1))
+			return 0;
+
+		odp_buffer_to_event_multi(buf, tbl, evv_size);
+
+		if (odp_unlikely(evv == ODP_EVENT_VECTOR_INVALID))
+			return 0;
+
+		odp_event_free_multi(tbl, evv_size);
+		odp_event_vector_free(evv);
+	}
```

Comment:
We are only interested about `odp_event_vector_alloc()` and `odp_event_vector_free()` performance here, so no need to fill the vectors.